core/net.go

```diff
@@ -209,7 +213,20 @@ func (n *OpenBazaarNode) SendCancel(peerId, orderId string) error {
 	}
 	err = n.Service.SendMessage(ctx, p, &m)
 	if err != nil {
-		if err := n.SendOfflineMessage(p, &m); err != nil {
+		//try to get public key from order
+		order, _, _, _, _, err := n.Datastore.Purchases().GetByOrderId(orderId)
+		var kp *libp2p.PubKey
+		if err != nil { //probably implies we can't find the order in the Datastore
+			kp = nil //instead SendOfflineMessage can try to get the key from the peerId
+		} else {
+			k, err := libp2p.UnmarshalPublicKey(order.GetVendorListings()[0].GetVendorID().GetPubkeys().Guid)
```

Comment:
SendCancel is called only by order.go:
n.SendCancel(contract.VendorListings[0].VendorID.Guid, orderId)
So VendorID is correct I believe.